src/channel.cr

```diff
@@ -97,11 +120,13 @@ class Channel(T)
 
       @senders.each do |sender|
         sender.state_ptr.value = DeliveryState::Closed
+        sender.select_context.try &.try_trigger
```

Comment:
I think this should be `next if (sender.select_context.try &.try_trigger) == false`

Reply:
why? If there are multiple selects waiting for the same channel you want all of them to unwait knowing the reason was that the underlying channel got closed.

Reply:
The reason is the issue #8300. Without `next`, the fiber whose select operation has been completed and not entered `unwait` stage can be enqueued into the scheduler.  This
is about unexpected fiber enqueue.

Reply:
Ok, I see.

I pushed some changes that I think makes more explicit when the fiber is enqueued (or not).
And I also added the spec for both receive and send.

Thanks!